geopandas/plotting.py

```diff
@@ -376,7 +376,9 @@ def plot_dataframe(df, column=None, cmap=None, color=None, ax=None, cax=None,
         Size of the resulting matplotlib.figure.Figure. If the argument
         axes is given explicitly, figsize is ignored.
     legend_kwds : dict (default None)
-        Keyword arguments to pass to ax.legend()
+        Keyword arguments to pass to matplotlib.pyplot.legend() in case of legend=True and
```

Comment:
This is true from our point of view, but you can also do `gdf.plot('data', scheme='quantiles', legend=True)`, which sets categorical to True, but user did not touch it so they might expect it defaulting to False. Can cause confusion. I would use simpler language here and say something along "Keyword arguments to pass to matplotlib.pyplot.legend() or matplotlib.pyplot.colorbar()".